### PR TITLE
authentication

### PR DESCRIPTION
**Pull request by kheersagar**

### CODE REVIEW

File: controllers/admin.js [hunk 1]
```diff
@@ -13,4 +18,66 @@ app.get("/", (req,res)=>{
...
+    const count = await User.estimatedDocumentCount()+1;
+    const {email,password} = req.body;
+    if(!(email,password)){
+      res.status(400).json({status:400,message:"Insufficient input",success:false});
+      res.redirect("/register");
+    }
+    const old = await User.findOne({email});
+    console.log(old);
+    if(old){
+      res.status(409).json({status:409,message:"User already exists",success:false,data:old});
```
Comment (kheersagar) [inline controllers/admin.js:32]: remove status and message success from the response where ever added

File: controllers/admin.js [hunk 2]
```diff
@@ -13,4 +18,66 @@ app.get("/", (req,res)=>{
...
+    res.status(500).json({status:500,message:"Internal server error",success:false,err:err});
+  }
+});
+
+app.post('/login', async(req,res)=>{
+  try{
+    const {email,password} = req.body;
+    if(!(email,password)){
+      res.status(400).json({status:400,message:"Insufficient input",success:false});
+      res.redirect("/login");
```
Comment (kheersagar) [inline controllers/admin.js:62]: redirection will happen from frontend no use from backend

File: app.js [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
+var session = require('express-session');
 const bodyParser = require("body-parser");
+var cookieParser = require('cookie-parser');
 app.use(bodyParser.urlencoded({extended: true}));
 
+// sessions set-up
+var sessionVar;
+const oneDay = 1000 * 60 * 60 * 24;
+app.use(session({
+    secret: "thisismysecrctekeyforthisnewproject",
```
Comment (kheersagar) [inline app.js:14]: add this in env

File: controllers/admin.js [hunk 4]
```diff
@@ -13,4 +18,66 @@ app.get("/", (req,res)=>{
   }
 });
 
+app.post("/register",async (req,res)=>{
+  try{
+    const count = await User.estimatedDocumentCount()+1;
+    const {email,password} = req.body;
+    if(!(email,password)){
+      res.status(400).json({status:400,message:"Insufficient input",success:false});
+      res.redirect("/register");
```
Comment (kheersagar) [inline controllers/admin.js:27]: remove redirect

File: controllers/admin.js [hunk 5]
```diff
@@ -13,4 +18,66 @@ app.get("/", (req,res)=>{
...
+    }
+    const old = await User.findOne({email});
+    console.log(old);
+    if(old){
+      res.status(409).json({status:409,message:"User already exists",success:false,data:old});
+    }
+    else{
+      let encryptedPassword = await bcrypt.hash(password, 10);
+      const user = await User.create({
+        uid:'A'+count,
```
Comment (kheersagar) [inline controllers/admin.js:37]: whats the use of uid

File: controllers/admin.js [hunk 6]
```diff
@@ -13,4 +18,66 @@ app.get("/", (req,res)=>{
...
+app.post('/login', async(req,res)=>{
+  try{
+    const {email,password} = req.body;
+    if(!(email,password)){
+      res.status(400).json({status:400,message:"Insufficient input",success:false});
+      res.redirect("/login");
+    }else{
+      const user = await User.findOne({email});
+      if (user && (await bcrypt.compare(password, user.password))) {
+      const token = jwt.sign(
```
Comment (kheersagar) [inline controllers/admin.js:66]: create this function as modular code instead of hardcoding

File: controllers/admin.js (hunk 1 above)
Comment (DaNtE-tk) [reply controllers/admin.js:32]: ADDRESSED

File: controllers/admin.js (hunk 2 above)
Comment (DaNtE-tk) [reply controllers/admin.js:62]: ADDRESSED

File: controllers/admin.js (hunk 4 above)
Comment (DaNtE-tk) [reply controllers/admin.js:27]: ADDRESSED

File: controllers/admin.js (hunk 5 above)
Comment (DaNtE-tk) [reply controllers/admin.js:37]: It work as a unique field for easy access of data from user records.

File: app.js (hunk 3 above)
Comment (DaNtE-tk) [reply app.js:14]: ADDRESSED

File: controllers/admin.js (hunk 5 above)
Comment (kheersagar) [reply controllers/admin.js:37]: we already have _id provided by mongodb

Comment (DaNtE-tk) [reply controllers/admin.js:37]: yes but, this one is dev customized...
should I remove it?

File: controllers/admin.js (hunk 6 above)
Comment (DaNtE-tk) [reply controllers/admin.js:66]: how do I do that? shall I create an external function for this?

File: controllers/admin.js (hunk 5 above)
Comment (kheersagar) [reply controllers/admin.js:37]: yes

File: controllers/admin.js (hunk 6 above)
Comment (kheersagar) [reply controllers/admin.js:66]: yes

File: controllers/admin.js (hunk 5 above)
Comment (DaNtE-tk) [reply controllers/admin.js:37]: ADDRESSED

**Pull request merged**